corehq/preindex/__init__.py

```diff
@@ -39,6 +40,12 @@ def get_designs(self):
         raise NotImplementedError()
 
     def sync_design_docs(self, temp=None):
+        if self.synced:
+            # workaround emit_post_sync_signal called twice
+            # https://code.djangoproject.com/ticket/17977
+            # this is to speed up test initialization
```

Comment:
I think that approach is solving a different problem than what this is working around. Note the title of linked ticket: "post_syncdb is triggered twice, in unittest, *even with dispatch_uid*." In this case two different places in the code are emitting the `emit_post_sync_signal` signal. Did I miss something?